webserver/views/api/v1/datasets.py

```diff
@@ -97,6 +97,165 @@ def delete_dataset(dataset_id):
+
+    :resheader Content-Type: *application/json*
+    """
+    raise NotImplementedError
+
+
+@bp_datasets.route("/<uuid:dataset_id>/classes", methods=["POST"])
+@auth_required
+def add_class(dataset_id):
+    """Add class into a dataset.
+
+    **Example request**:
+
+    .. sourcecode:: json
+
+        {
+            "name": "Not Mood",
+            "description": "Dataset for mood misclassification.",
+        }
+
+    :reqheader Content-Type: *application/json*
+    :<json string name: *Required.* Name of the class. Must be unique within a dataset.
+    :<json string description: *Optional.* Description of the class.
+    :<json array public: *Optional.* Array of recording MBIDs (``string``) to add into that class. For example:
+        ``["770cc467-8dde-4d22-bc4c-a42f91e"]``.
+
+
+    :resheader Content-Type: *application/json*
+    """
+    raise NotImplementedError
+
+
+@bp_datasets.route("/<uuid:dataset_id>/classes", methods=["PUT"])
+@auth_required
+def update_class(dataset_id):
+    """Update class in a dataset.
+
+    If one of the fields is not specified, it will not be updated.
+
+    **Example request**:
+
+    .. sourcecode:: json
+
+        {
+            "name": "Very happy",
+            "new_name": "Recordings that represent ultimate happiness."
+        }
+
+    :reqheader Content-Type: *application/json*
+    :<json string name: *Required.* Current name of the class.
+    :<json string new_name: *Optional.* New name of the class. Must be unique within a dataset.
+    :<json string description: *Optional.* Description of the class.
+
+    :resheader Content-Type: *application/json*
+    """
+    raise NotImplementedError
+
+
+@bp_datasets.route("/<uuid:dataset_id>/classes", methods=["DELETE"])
+@auth_required
+def delete_class(dataset_id):
+    """Delete class from a dataset.
+
+    **Example request**:
+
+    .. sourcecode:: json
+
+        {
+            "name": "Sad"
+        }
+
+    :reqheader Content-Type: *application/json*
+    :<json string name: *Required.* Name of the class.
+
+    :resheader Content-Type: *application/json*
+    """
+    raise NotImplementedError
+
+
+@bp_datasets.route("/<uuid:dataset_id>/recordings", methods=["PUT"])
+@auth_required
+def add_recordings(dataset_id):
+    """Add recordings to a class in a dataset.
+
+    **Example request**:
+
+    .. sourcecode:: json
+
+        {
+            "class_name": "Happy",
+            "recordings": ["770cc467-8dde-4d22-bc4c-a42f91e"]
+        }
+
+    :reqheader Content-Type: *application/json*
+    :<json string class_name: *Required.* Name of the class.
+    :<json array recordings: *Required.* Array of recoding MBIDs (``string``) to add into that class.
+
+    :resheader Content-Type: *application/json*
+    """
+
+    #The json in this method's example request is given to be 'class_name' instead of 'name' which is in other json requests.
+
+    ds = get_dataset(dataset_id)
+    db.dataset.add_recordings(ds)
```

Comment:
This isn't doing the correct task.

In URL endpoints we can get data from two places. From the URL, and from the request payload.
In this case with a url such as
`/api/v1/datasets/b86df890-32e0-4550-bc1b-e6e99414c135/recordings`
the `route` function will extract part of the url and put it in the `dataset_id` variable

The request payload is sent separately, and you access it with `request.get_json()`: https://github.com/vibhorbajpai/acousticbrainz-server/blob/6bd08b330251e0a1cf7a9d06280e84db66370bf6/webserver/views/api/v1/datasets.py#L68

You appear to have confused the request payload with the data which already exists in the database. In this case, you are getting the contents of a dataset and then using it to call the `db.dataset.add_recordings` method. These dictionaries are different, so I don't think that the `add_recordings` method will work.

The response of `get_dataset` is something like this
```json
{
  "author": 1,
  "classes": [
    {"description": null,
     "id": "106",
     "name": "classthree",
     "recordings": ["075a3b50-4650-4fb6-8c99-66e1af7bb07e"]
    }],
  "created": "Thu, 23 Jun 2016 13:36:01 GMT",
  "description": "",
  "id": "f67afc32-e417-4304-9ed9-8eb41a800dc8",
  ... other data removed ...
}
```

and the format which you expect has `recordings` and `class_name` fields: https://github.com/vibhorbajpai/acousticbrainz-server/blob/6bd08b330251e0a1cf7a9d06280e84db66370bf6/db/dataset.py#L65

This method should instead do something like get the dataset, get the request body, validate that the format of the body is correct, and call `add_recordings` passing in both the dataset id and the request body.

In these methods you have to check that the dataset exists and that the current user has permission to edit this dataset. Instead of `get_dataset` we have a helper which you can use, `get_check_dataset`